table/block_based/block_based_table_builder.cc

```diff
@@ -1031,6 +1029,13 @@ void BlockBasedTableBuilder::WriteBlock(const Slice& raw_block_contents,
   if (!ok()) {
     return;
   }
+
+  if (is_data_block && r->table_options.prepopulate_block_cache) {
+    handle->set_offset(r->get_offset());
+    handle->set_size(block_contents.size());
+    InsertBlockInCache(raw_block_contents, kNoCompression, handle);
```

Comment:
```InsertBlockInCache``` is only for uncompressed blocks right? The 2nd argument seems redundant.

Reply:
Yes. So that check is added in the ```InsertBlockInCache``` function. That's why I am passing the compression type instead of checking at caller site.

Reply:
Ah got it. I will remove it. The check is not needed.